src/braket/experimental/autoqasm/program/program.py

```diff
@@ -88,6 +88,7 @@ def __init__(self, user_config: Optional[UserConfig] = None):
         self.return_variable = None
         self._qubits_seen = set()
         self._var_idx = 0
+        self._in_main = False
```

Comment:
Just curious, could you check that `self.subroutines_processing` is empty, instead of introducing a new variable for this? Or does `subroutines_processing` behave differently than what you need?

Reply:
Thank you!! This worked great